tests/DefaultUnleashTest.php

```diff
@@ -813,6 +826,182 @@ public function testAutoRegistrationOnInvalidResponse()
         );
     }
 
+    public function testDependenciesUnresolvedFeatureVariants()
+    {
+        $data = [
+            'features' => [
+                [
+                    'name' => 'child',
+                    'enabled' => true,
+                    'strategies' => [
+                        [
+                            'name' => 'default',
+                        ],
+                    ],
+                    'dependencies' => [
+                        [
+                            'feature' => 'parent',
+                            'variants' => ['variant1'],
+                        ],
+                    ],
+                ],
+                [
+                    'name' => 'parent',
+                    'enabled' => true,
+                    'strategies' => [
+                        [
+                            'name' => 'default',
+                        ],
+                    ],
+                    'variants' => [
+                        [
+                            'name' => 'variant1',
+                            'weight' => 1,
+                        ],
+                    ],
+                ],
+            ],
+        ];
+        $this->pushResponse($data);
+        $instance = $this->getCachedInstance();
+        self::assertTrue($instance->isEnabled('child'));
+    }
+
+    public function testDependenciesDirectFeatureVariants()
```

Comment:
This test might be worth adding to client specs.

Reply:
I though "Child is enabled if its parent matches a specific variant." matches this case. Can you elaborate what exactly are we missing in the client spec?

Reply:
If I'm not mistaken, the spec tests only contain variants in strategies, not variants in features directly.

Reply:
That's correct since variants and strategy variants have separate spec that should prove that they behave the same. In the long term we'd like to migrate towards strategy variants and slowly sunset feature environment variants.